Assignment4/AlvaroAlonso-0-b190300/task07.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1tV5j-DRcpPtoJGoMj8v2DSqR_9wyXeiE
+
+**Task 07: Querying RDF(s)**
+"""
+
+#pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2020-2021/master/Assignment4"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/resources/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+from rdflib.plugins.sparql import prepareQuery
+
+ns = Namespace("http://somewhere#")
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    ?Subject rdfs:subClassOf ns:Person
+  }
+  ''',
+  initNs = { "ns": Namespace("http://somewhere#"), "rdfs": RDFS}
+)
+# Visualize the results
+# RDFLib
+for s,p,o in g.triples((None,RDFS.subClassOf,ns.Person)):
+  print(s)
+
+# SPARQL
+for r in g.query(q1):
+  print(r.Subject)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**"""
+q2 = prepareQuery('''
+  SELECT DISTINCT ?Subject WHERE { 
+    {?Subject rdf:type ns:Person. }
+    UNION 
+    {?s rdfs:subClassOf ns:Person.
```

Comment:
use chained properties with * to get the full taxonomy of classes.

Reply:
Just uploaded a new version.